src/functions-and-arrays.js

```diff
@@ -1,41 +1,148 @@
-
+function findLongestWord(array) {
+  if (!array || array.length === 0) return null;
+  let longestWord = '';
+  // for (let index = 0; index < array.length; index++) {
+  //   const element = array[index];
+  //   if (element.length > longestWord.length) longestWord = element;
+  // }
+  array.forEach((element) => {
+    if (element.length > longestWord.length) longestWord = element;
+  });
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(numbers) {
+  if (!numbers || numbers.length === 0) return 0;
+  let sum = 0;
+  numbers.forEach((num) => {
+    sum += num;
+  });
+  return sum;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
+function sum(arr) {
+  if (!arr || arr.length === 0) return 0;
+  let sum = 0;
+  arr.forEach((num) => {
+    switch (typeof num) {
+      case 'number':
+        sum += num;
+        break;
+      case 'string':
+        sum += num.length;
+        break;
+      case 'boolean':
+        if (num) sum += 1;
+        else sum += 0;
+        break;
+      default:
+        () => {
+          throw new Error(`Unsupported data type sir or ma'am`);
+        };
+    }
+  });
+  return sum;
+}
+// BONUS - Iteration #3.1: A generic sum() function
+function sum(arr) {
+  if (!arr) return null;
+
+  let sum = 0;
+  for (let el of arr) {
+    // usamos la condición "typeof" para reusar esta función en la iteración siguiente (4.2)
+    // y poder calcular directamente la suma de carácteres en el array de words
+    if (typeof el === 'string') sum += el.length;
+    else if (typeof el === 'object') throw new Error(`Unsupported data type sir or ma'am`);
+    else sum += el;
+  }
+  return sum;
+}
 
+// console.log(sum([6, 12, 'miami', 1, 'barca', '200', 'lisboa', 8, [], {}]));
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbers) {
+  if (!numbers || numbers.length === 0) return null;
+  let sum = 0;
+  numbers.forEach((num) => {
+    sum += num;
+  });
+  return sum / numbers.length;
+}
 
+// console.log(averageNumbers(numbersAvg));
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(words) {
+  if (!words || words.length === 0) return null;
+  let sum = 0;
+  words.forEach((num) => {
+    sum += num.length;
+  });
+  return sum / words.length;
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(arr) {
+  if (!arr || arr.length === 0) return null;
+  let sum = 0;
+  arr.forEach((num) => {
+    switch (typeof num) {
+      case 'number':
+        sum += num;
+        break;
+      case 'string':
+        sum += num.length;
+        break;
+      case 'boolean':
+        if (num) sum += 1;
+        else sum += 0;
```

Comment:
Hola Ale la verdad no queria quedarme con la espina, estuve checando y el error es el break, hizo falta aquí puedes quitar la función anonima y pasa ya que como no estaba el break si ejecutaba el error 

```suggestion
break;
        else sum += 0;
```